include/openthread/tcp.h

```diff
@@ -395,17 +395,23 @@ otError otTcpBind(otTcpEndpoint *aEndpoint, const otSockAddr *aSockName);
  */
 enum
 {
-    OT_TCP_CONNECT_NO_FAST_OPEN = 1 << 0,
+    OT_TCP_CONNECT_DISABLE_FAST_OPEN = 1 << 0,
+    OT_TCP_CONNECT_ENABLE_FAST_OPEN  = 1 << 1,
```

Comment:
What does it mean if we have both `1<<0` and `1<<1` bits set in a given flag?

How about using a single bit-flag to indicate "fast-open" status? The bit-flag would be set to enable fast-open, and cleared to disable it.

Reply:
OK --- I reverted this to the old API with the single `OT_TCP_CONNECT_NO_FAST_OPEN` flag.

Whether TCP Fast Open should be enabled by default or not is a point worth additional discussion. The argument for having TFO enabled by default is that it encourages programmers to use the feature and provides performance closer to a UDP-based solution. @StuartCheshire prefers the existing API for this reason. I'm happy to change the API if there's a consensus on having TFO disabled by default but let's discuss that carefully first.